tests/test_predict.py

```diff
@@ -69,3 +73,63 @@ def test_predict(model_class, env_id, device):
 
         action, _ = model.predict(vec_env_obs, deterministic=False)
         assert action.shape[0] == vec_env_obs.shape[0]
+
+
+class FlattenBatchNormExtractor(BaseFeaturesExtractor):
+    """
+    Feature extract that flatten the input and uses batch normalization.
+    Used as a placeholder when feature extraction is not needed.
+
+    :param observation_space:
+    """
+
+    def __init__(self, observation_space: gym.Space, batch_norm_elements=4):
+        super(FlattenBatchNormExtractor, self).__init__(observation_space, get_flattened_obs_dim(observation_space))
+        self.flatten = nn.Flatten()
+        self.batch_norm = nn.BatchNorm1d(batch_norm_elements)
+        self.dropout = nn.Dropout(0.5)
+
+    def forward(self, observations: th.Tensor) -> th.Tensor:
+        result = self.flatten(observations)
+        result = self.batch_norm(result)
+        result = self.dropout(result)
+        return result
+
+
+@pytest.mark.parametrize("model_class", MODEL_LIST)
+@pytest.mark.parametrize("env_id", ["Pendulum-v0", "CartPole-v1"])
+@pytest.mark.parametrize("device", ["cpu", "cuda", "auto"])
+def test_batch_norm_dropout(model_class, env_id, device):
+    if device == "cuda" and not th.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    features_extractor_kwargs = dict(batch_norm_elements=3)
+
+    if env_id == "CartPole-v1":
+        features_extractor_kwargs = dict(batch_norm_elements=4)
+
+        if model_class in [SAC, TD3]:
+            return
+    elif model_class in [DQN]:
+        return
+
+    model_kwargs = dict()
+
+    if model_class in [DQN, TD3, SAC]:
```

Comment:
I saw that you modified only the on policy algorithm `train()` method, so currently, the parameters of batch norm are probably never changed for them (you should probably add a test to check that something was learned).